db/view/view_update_generator.cc

```diff
@@ -39,10 +39,22 @@ future<> view_update_generator::start() {
                 if (result == stop_iteration::yes) {
                     break;
                 }
-                entry.t->move_sstable_from_staging_in_thread(entry.sst);
-                _registration_sem.signal();
+                auto t = entry.t.get();
+                auto it = _sstables_to_move.find(t);
+                if (it == _sstables_to_move.end()) {
+                    _sstables_to_move.emplace(t, entry);
+                } else {
+                    it->second.sstables.push_back(entry.sst);
+                }
```

Comment:
Wouldn't `_sstables_to_move[t].push_back(entry.sst`) achieve the same?

Reply:
Yeah, with a default constructor for `sstables_to_move`.
Good idea...

Reply:
Hmm, we also need to set `_sstables_to_move[t].t = entry.t` at least once
so maybe it's more elegant but less efficient.
I want to hold on to a lw_shared_ptr<table> rather than just a `table*`.
If it would be the latter case, we could just use the `table*` key.

Reply:
done